src/jdk.compiler/share/classes/com/sun/tools/javac/processing/PrintingProcessor.java

```diff
@@ -770,5 +784,16 @@ private void indent() {
             writer.print(spaces[indentation]);
         }
 
+        private boolean isImportantType(TypeMirror type) {
+            if (!type.getAnnotationMirrors().isEmpty()) {
+                return true;
+            }
+            TypeElement e2 = (TypeElement)
+                ((DeclaredType) type).asElement();
+            if (!e2.getKind().isClass()) {
+                return true;
+            }
+            return e2.getSuperclass().getKind() != TypeKind.NONE;
```

Comment:
I think it would be helpful to include some textual description of the intended semantics here including

"Omit the bound is unannotated java.lang.Object."

Reply:
Done:
https://github.com/openjdk/jdk/pull/25002/commits/9bb3677511b936002a22dde7401ba01ed116a2a8